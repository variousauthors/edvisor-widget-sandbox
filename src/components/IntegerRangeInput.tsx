import * as React from 'react';
import Select from './Select';

interface IntegerRangeInputProps {
  boundsOptions: { value: string, name: string }[],
  onChange: (value: any) => void,
  initialValue?: string,
  initialBounds?: string,
  vocab?: { value: string, bounds: string },
}

export default function IntegerRangeInput ({ boundsOptions, onChange, initialValue = "", initialBounds = undefined, vocab={ value: "value", bounds: "bounds" } }: IntegerRangeInputProps) {

  return (
    <div>
      <label>Age: </label>
      <input 
        type='number' 
        min='0'
        step='1'
        value={ initialValue }
        onChange={ (e) => onChange({ [vocab.value]: e.target.value })  }/>
      <Select 
        defaultValue={ initialBounds } 
        data={ boundsOptions } 
        onChange={ (value) => onChange({ [vocab.bounds]: value }) } />
    </div>
  );
}

