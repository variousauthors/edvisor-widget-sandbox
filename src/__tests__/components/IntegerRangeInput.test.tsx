import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import IntegerRangeInput from '../../components/IntegerRangeInput';

let nothing = (value) => {};
let boundsOptions = [
  { value: 'lte', name: 'less than' },
  { value: 'gte', name: 'greater than' },
  { value: 'any', name: 'any' },
]

let vocab = {
  value: "bob", bounds: "jill"
}

describe('<IntegerRangeInput />', () => {
  it('renders', () => {

    const tree = renderer.create(<IntegerRangeInput boundsOptions={ boundsOptions } onChange={ nothing } />).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('uses the correct constructs the bounds object onChange', ()=> {

    let onChange = (value) => { expect(value.bounds).toBe(1); };
    const wrapper = shallow(<IntegerRangeInput boundsOptions={ boundsOptions } onChange={ onChange } />);

    wrapper.find('Select').simulate('change', 1);
  });

  it('correctly maps based on the provided vocabulary', ()=> {

    let onChange = (value) => { expect(value.jill).toBe(1); };
    const wrapper = shallow(<IntegerRangeInput vocab={ vocab } boundsOptions={ boundsOptions } onChange={ onChange } />);

    wrapper.find('Select').simulate('change', 1);
  });
});
