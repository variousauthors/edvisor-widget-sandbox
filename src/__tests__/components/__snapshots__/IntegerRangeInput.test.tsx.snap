// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntegerRangeInput /> renders 1`] = `
<div>
  <label>
    Age: 
  </label>
  <input
    min="0"
    onChange={[Function]}
    step="1"
    type="number"
    value=""
  />
  <select
    defaultValue={undefined}
    onChange={[Function]}
  >
    <option
      value="lte"
    >
      less than
    </option>
    <option
      value="gte"
    >
      greater than
    </option>
    <option
      value="any"
    >
      any
    </option>
  </select>
</div>
`;
